app/models/mvi.rb

```diff
@@ -120,11 +121,23 @@ def mvi_response
 
   def response_from_redis_or_service
     do_cached_with(key: @user.uuid) do
-      mvi_service.find_profile(@user)
+      profile = mvi_service.find_profile(@user)
+      self.class.redis_namespace_ttl = set_ttl(profile)
+      profile
```

Comment:
`redis_namespace_ttl` is a class var that's being modified through an instance method, so I think it's a thread safety issue. Granted, I guess the worst thing that could happen would be a failed response saved for `each_ttl` or a successful response stored for `failure_ttl`. Not all that dangerous for right now, but just an observation.

Reply:
Thanks for pointing this out @omgitsbillryan I'll keep an eye on things and see if it becomes a problem.